resource-manager/src/main/java/org/ihtsdo/otf/resourcemanager/ResourceManager.java

```diff
@@ -1,113 +1,353 @@
 package org.ihtsdo.otf.resourcemanager;
 
+import com.amazonaws.services.s3.AmazonS3;
+import com.amazonaws.services.s3.AmazonS3ClientBuilder;
 import com.amazonaws.services.s3.model.AmazonS3Exception;
-import org.springframework.cloud.aws.core.io.s3.PathMatchingSimpleStorageResourcePatternResolver;
+import org.ihtsdo.otf.resourcemanager.ResourceConfiguration.Cloud;
 import org.springframework.core.io.FileSystemResourceLoader;
 import org.springframework.core.io.Resource;
 import org.springframework.core.io.ResourceLoader;
 import org.springframework.core.io.WritableResource;
 import org.springframework.util.StreamUtils;
 
+import java.io.File;
 import java.io.IOException;
 import java.io.InputStream;
 import java.io.OutputStream;
+import java.nio.file.Files;
+import java.nio.file.StandardCopyOption;
+import java.util.Objects;
 
 /**
- * Service to read and write file resources from S3 or local disk.
- * Use spring-cloud-aws-autoconfigure dependency to autowire an S3 capable ResourceLoader into your configuration.
+ * Service to read, write, delete and move file resources from S3 or local
+ * disk. Use spring-cloud-aws-autoconfigure dependency to autowire an S3 capable
+ * ResourceLoader into your configuration.
  */
 public class ResourceManager {
 
 	private final ResourceConfiguration resourceConfiguration;
 	private final ResourceLoader resourceLoader;
+	private AmazonS3 amazonS3;
 
-	public ResourceManager(ResourceConfiguration resourceConfiguration, ResourceLoader cloudResourceLoader) {
-		this.resourceConfiguration = resourceConfiguration;
+	/**
+	 * Creates the {@link ResourceManager} with the corresponding <code>resourceConfiguration</code>
+	 * and <code>cloudResourceLoader</code>. {@code NullPointerException} is thrown if
```

Comment:
A null pointer will not be thrown if the resourceConfiguration is not cloud and the cloudResourceLoader is null. This seems like an acceptable situation for unit testing when using this project as a library.

Reply:
Fixed